### DISCUSSION TITLE
Cancellation of order

### DISCUSSION THREAD

**Discussion by BLR-0118** [beckn/protocol-specifications#238] [Core Specification]
If BAP wants to cancel an order, they can call /cancel endpoint on the BPP, for which they will get the /on_cancel response from the BPP. How can the BPP cancel an order? Is there a way for the BPP to cancel an order and have the BAP notified?

**Comment by ravi-prakash-v** [comment]
Yes, when the BPP wants to cancel an order, it just sends an unsolicited `on_cancel` request to the BAP with the order.state as `CANCELLED`
